src/queries/__tests__/a11yHint.test.tsx

```diff
@@ -38,55 +38,95 @@ const Section = () => (
-  expect(asyncButton.props.accessibilityHint).toEqual(BUTTON_HINT);
-  await expect(findByA11yHint(NO_MATCHES_TEXT)).rejects.toThrow(
-    getNoInstancesFoundMessage('accessibilityHint')
-  );
-
-  await expect(findByA11yHint(TEXT_HINT)).rejects.toThrow(
-    getMultipleInstancesFoundMessage('accessibilityHint', TEXT_HINT)
-  );
+describe('getByA11yHint, queryByA11yHint, findByA11yHint', () => {
+  test('finds the element by a11y hint', async () => {
+    const { getByA11yHint, queryByA11yHint, findByA11yHint } = render(
+      <Section />
+    );
+
+    expect(getByA11yHint(BUTTON_HINT).props.accessibilityHint).toEqual(
+      BUTTON_HINT
+    );
+    const button = queryByA11yHint(BUTTON_HINT);
+    expect(button?.props.accessibilityHint).toEqual(BUTTON_HINT);
+
+    const asyncButton = await findByA11yHint(BUTTON_HINT);
+    expect(asyncButton.props.accessibilityHint).toEqual(BUTTON_HINT);
+  });
+
+  test('thow error multiple instances found when multiple components match the hint', async () => {
+    const { getByA11yHint, queryByA11yHint, findByA11yHint } = render(
+      <Section />
+    );
+
+    expect(() => getByA11yHint(TEXT_HINT)).toThrow(
+      getMultipleInstancesFoundMessage('accessibilityHint', TEXT_HINT)
+    );
+    expect(() => queryByA11yHint(TEXT_HINT)).toThrow(
+      getMultipleInstancesFoundMessage('accessibilityHint', TEXT_HINT)
+    );
+
+    await expect(findByA11yHint(TEXT_HINT)).rejects.toThrow(
+      getMultipleInstancesFoundMessage('accessibilityHint', TEXT_HINT)
+    );
+  });
+
+  test('getByA11yHint, findByA11yHint throw error not found when no element matches the hint', async () => {
+    const { getByA11yHint, findByA11yHint } = render(<Section />);
+
+    expect(() => getByA11yHint(NO_MATCHES_TEXT)).toThrow(
+      getNoInstancesFoundMessage('accessibilityHint')
+    );
+
+    await expect(findByA11yHint(NO_MATCHES_TEXT)).rejects.toThrow(
+      getNoInstancesFoundMessage('accessibilityHint')
+    );
+  });
+
+  test('queryByA11yHint returns null when no element matches the a11y hint', async () => {
+    const { queryByA11yHint } = render(<Section />);
+
+    expect(queryByA11yHint(NO_MATCHES_TEXT)).toBeNull();
+  });
 });
 
-test('getAllByA11yHint, queryAllByA11yHint, findAllByA11yHint', async () => {
-  const { getAllByA11yHint, queryAllByA11yHint, findAllByA11yHint } = render(
-    <Section />
-  );
-
-  expect(getAllByA11yHint(TEXT_HINT)).toHaveLength(2);
-  expect(queryAllByA11yHint(TEXT_HINT)).toHaveLength(2);
-
-  expect(() => getAllByA11yHint(NO_MATCHES_TEXT)).toThrow(
-    getNoInstancesFoundMessage('accessibilityHint')
-  );
-  expect(queryAllByA11yHint(NO_MATCHES_TEXT)).toEqual([]);
-
-  await expect(findAllByA11yHint(TEXT_HINT)).resolves.toHaveLength(2);
-  await expect(findAllByA11yHint(NO_MATCHES_TEXT)).rejects.toThrow(
-    getNoInstancesFoundMessage('accessibilityHint')
-  );
+describe('getAllByA11yHint, queryAllByA11yHint, findAllByA11yHint', () => {
```

Comment:
Pls skip the describe block, and just use top-level `test`, e.g. `byA11yHint finds the element by a11y hint`

Reply:
yes, I removed the commit that reorganized the queries by test case (with the describes) will do that in the follow up POUR